### PR TITLE
crimson/os/seastore/btree: interrupt transactions immediately when getting children

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
jenkins test make check

**Review by cyx1231st — APPROVED** [review]
I see, the child may be freed if the transaction is invalidated at `get_child_fut()`.

File: src/crimson/os/seastore/btree/fixed_kv_btree.h [hunk 1]
```diff
@@ -1491,9 +1491,9 @@ class FixedKVBtree {
     // checking the lba child must be atomic with creating
     // and linking the absent child
     if (v.has_child()) {
-      return v.get_child_fut().safe_then(
-        [on_found=std::move(on_found), node_iter, c,
-        parent_entry](auto child) mutable {
+      return trans_intr::make_interruptible(std::move(v.get_child_fut())
+      ).si_then([on_found=std::move(on_found), node_iter, c,
+                parent_entry](auto child) mutable {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/btree/fixed_kv_btree.h:1496]: Wondering whether `mutable` is needed for `on_found`, it doesn't look modified, same to the below places.

**Comment by cyx1231st** [comment]
Thanks @xxhdx1985126! Is the comment https://github.com/ceph/ceph/pull/58357#discussion_r1660408486 make sense to you?

**Comment by xxhdx1985126** [comment]
> Thanks @xxhdx1985126 to link the tracker! Is the comment [#58357 (comment)](https://github.com/ceph/ceph/pull/58357#discussion_r1660408486) make sense to you?

Yeah, sorry, I forgot to address that comment, please take a look again, thanks:)

**Comment by cyx1231st** [comment]
https://shaman.ceph.com/builds/ceph/wip-yingxin-testing-2024-07-11-1607/

**Comment by cyx1231st** [comment]
https://pulpito.ceph.com/yingxin-2024-07-12_03:37:20-crimson-rados-wip-yingxin-testing-2024-07-12-0957-distro-default-smithi/

**Comment by tchaikov** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
Test results:
* 7798503, 7798504: https://tracker.ceph.com/issues/66761
* 7798487, 7798488: https://tracker.ceph.com/issues/63844
* -7798506: https://tracker.ceph.com/issues/66924 (new, but looks unrelated)

**Comment by cyx1231st** [comment]
jenkins test api

**Pull request merged**